Docs cleanups, clarifications, and fixes (#334)
akka/akka-management: docs/src/main/paradox/bootstrap/index.md

<a id="bootstrap"></a>
# Akka Cluster Bootstrap

Akka Cluster Bootstrap helps forming (or joining to) a cluster by using @ref:[Akka Discovery](../discovery/index.md)

> to discover peer nodes.  It is an alternative to configuring  static `seed-nodes` in common dynamic deployment environments

Reviewer: ```suggestion
to discover peer nodes.  It is an alternative to configuring static `seed-nodes` in common dynamic deployment environments
```

> like on kubernetes or on AWS.

Reviewer: ```suggestion
such as on Kubernetes or AWS.
```

> It builds on the flexibility of Akka Discovery to make use of a range of different discovery mechanisms depending on the

Reviewer: ```suggestion
It builds on the flexibility of Akka Discovery, leveraging a range of discovery mechanisms depending on the
```

* During bootstrap each node queries service discovery repeatedly to get the initial contact points until at least the
  number of contact points as defined in `contact-point-discovery.required-contact-point-nr` has been found.
* Each node then probes the found contact points' `/bootstrap/seed-nodes` endpoint to see if a cluster has already
  been formed
    * If there is an existing cluster, it joins the cluster and bootstrapping is finished.
    * If no cluster exists, each node returns an empty list of `seed-nodes`. In that case, the node with the lowest
      address from the set of contact points forms a new cluster and starts advertising itself as a seed node.
* Other nodes will start to see the `/bootstrap/seed-nodes` of the node that self-joined and will join its cluster.

See @ref[full bootstrap process and advanced configuration](details.md) for more details on the process.

## Joining Mechanism Precedence

As Akka Cluster allows nodes to join to a cluster using a few methods, the precedence of each method
is strictly defined and is as follows:

- If `akka.cluster.seed-nodes` (in your `application.conf`) are non-empty, those nodes will be joined, and bootstrap

>   will NOT execute even if `start()` is called (a warning will be logged, though).

Reviewer: ```suggestion
  will NOT execute even if `start()` is called, however a warning will be logged.
```

> - If an explicit `cluster.join` or `cluster.joinSeedNodes` is invoked before the bootstrap completes that

Reviewer: ```suggestion
- If an explicit `cluster.join` or `cluster.joinSeedNodes` is invoked before the bootstrap completes, that
```

  joining would take precedence over the bootstrap (but it's not recommended to do so, see below).
- The Cluster Bootstrap mechanism takes some time to complete, but eventually issues a `joinSeednodes`.

@@@ warning
  It is NOT recommended to mix various joining mechanisms. Pick one mechanism and stick to it in order to
  avoid any surprises during cluster formation. E.g. do NOT set `akka.cluster.seed-nodes` if you are going
  to be using the Bootstrap mechanism.
@@@

## Bootstrap Recipes

To see how to configure and use bootstrap in various environments such as Kubernetes, see @ref[recipes](recipes.md).

> @@@ index

Reviewer 2: :+1:
